ginjwt/helpers.go

```diff
@@ -0,0 +1,63 @@
+package ginjwt
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// RegisterViperOIDCFlags ensures that the given Viper and cobra.Command instances
+// have the following command line/configuration flags registered:
+//
+// * `oidc`: Enables/disables OIDC Authentication
+// * `oidc-aud`: Specifies the expected audience for the JWT token
+// * `oidc-issuer`: Specifies the expected issuer for the JWT token
+// * `oidc-jwksuri`: Specifies the JSON Web Key Set (JWKS) URI.
+// * `oidc-roles-claim`: Specifies the roles to be accepted for the JWT claim.
+// * `oidc-username-claim`: Specifies a username to use for the JWT claim
+//
+// A call to this would normally look as follows:
+//
+//		ginjwt.RegisterViperOIDCFlags(viper.GetViper(), serveCmd)
+//
+func RegisterViperOIDCFlags(v *viper.Viper, cmd *cobra.Command) {
+	cmd.Flags().Bool("oidc", true, "use oidc auth")
+	ViperBindFlag("oidc.enabled", cmd.Flags().Lookup("oidc"))
+	cmd.Flags().String("oidc-aud", "", "expected audience on OIDC JWT")
+	ViperBindFlag("oidc.audience", cmd.Flags().Lookup("oidc-aud"))
+	cmd.Flags().String("oidc-issuer", "", "expected issuer of OIDC JWT")
+	ViperBindFlag("oidc.issuer", cmd.Flags().Lookup("oidc-issuer"))
+	cmd.Flags().String("oidc-jwksuri", "", "URI for JWKS listing for JWTs")
+	ViperBindFlag("oidc.jwksuri", cmd.Flags().Lookup("oidc-jwksuri"))
+	cmd.Flags().String("oidc-roles-claim", "claim", "field containing the permissions of an OIDC JWT")
+	ViperBindFlag("oidc.claims.roles", cmd.Flags().Lookup("oidc-roles-claim"))
+	cmd.Flags().String("oidc-username-claim", "", "additional fields to output in logs from the JWT token, ex (email)")
+	ViperBindFlag("oidc.claims.username", cmd.Flags().Lookup("oidc-username-claim"))
+}
+
+// GetAuthConfigFromFlags builds an AuthConfig object from flags provided by
+// the viper tooling. This utility function assumes that the
+// `RegisterViperOIDCFlags` function was called beforehand.
+//
+// A call to this would normally look as follows:
+//
+//		ginjwt.GetAuthConfigFromFlags(viper.GetViper())
+func GetAuthConfigFromFlags(v *viper.Viper) AuthConfig {
+	return AuthConfig{
+		Enabled:       v.GetBool("oidc.enabled"),
+		Audience:      v.GetString("oidc.audience"),
+		Issuer:        v.GetString("oidc.issuer"),
+		JWKSURI:       v.GetString("oidc.jwksuri"),
+		LogFields:     v.GetStringSlice("oidc.log"),
```

Comment:
You have log fields here but not in the idc struck above, otherwise I think this will be useful

Reply:
I actually wanted to ask about this! I posted my question here: https://github.com/packethost/bender/pull/17 but never did get an answer. I saw this same pattern in archetype service and bender... where is this coming from usually? I never saw it defined in either code-bases.